### ISSUE TITLE
Text Embeddings by Weakly-Supervised Contrastive Pre-training (2022.12)

### ISSUE THREAD

**Issue by AtsunoriFujita** [AtsunoriFujita/paper-reading-notes#3]
# Text Embeddings by Weakly-Supervised Contrastive Pre-training

https://arxiv.org/abs/2212.03533

## 問題意識

- 言語モデル（BERT, T5など）を限られた数のラベル付きデータセット（SNLI, MS-MARCO）でFine-tuningする先行研究は短いセンテンスの埋め込み表現に焦点が当てられていて、長い文書での問題はそのままになっている。
- 言語モデルそのものは埋め込みモデルに理想的とは言えないので、埋め込みモデルを事前学習するアプローチも増えている。しかし、大量の低品質生成データで事前学習したモデルでは、BM25のような古典アプローチに勝てないケースもある。

## 本論文のアプローチ

- Zero-shot, Few-shotで使える汎用の事前学習済み埋め込みモデル、E5（**E**mb**E**ddings from
bidir**E**ctional **E**ncoder r**E**presentations）を提案。
- 大規模に収集したテキストペアデータセット、CCPairsデータセットを構築。
    - Redditの投稿とコメント、Stackexchangeの質問と最も投票された回答、英語Wikipediaのエンティティ名+セクションタイトルとパッセージ、Common Crawlのタイトルとパッセージなどが含まれる。クリーニング後のペア数は1.3BでそのほとんどはRedditとCommon Crawlから来ている。
    - Consistency-based filter: 一度、1.3Bで学習したモデルでそれぞれのテキストペアのパッセージを100万件のランダム抽出したパッセージの中で検索。top k（論文ではk=2）に入らないペアは品質が低いと判断した。その結果、約270Mのテキストペアとなった。
- モデルの事前学習はCCpairsを使った対照学習。損失関数はInfoNCE loss。質問とパッセージがどちらでも問題ないデータソースはランダムに入れ替え。学習中の負例の選択方法はin-batch negatives（バッチ内の他のペアからのパッセージを使う）。MiniLM（E5 small）, BERT base（E5 base）, BERT large（E5 large）に基づいたモデルでBS=32,768（負例を多く含めたい）、20k steps（2.5 epoch相当）学習。V100（16, 32, 64枚）で1-2日程度。Gradient checkpointing, Mixed Precisionを使用。
- パフォーマンスをさらに向上させるためにラベル付きデータセットでFine-tuningをした。データセットはSNLI, MS-MARCO, NQ。BS=256で3 epochs、8 GPUsで学習。

## 評価とAblation Study

- BEIRベンチマークで評価。Fine-tuningをしていない事前学習済みモデルでもほとんどのデータセットでBM25や既存の手法に勝った。Fine-tuning済みではさらに性能が向上。MTEBベンチマークでもパラメータが300MしかないE5 largeで4.8BのSentence-T5 xxlなどに匹敵するパフォーマンスであった。
- BERT+Fine-tuningと比較したが、CCpairsデータセットで事前学習している方が性能が良かった。
- 負例を増やすためにバッチサイズを大きくするアプローチは有効だった。
- ダウンストリームのタスクによって、最良の結果が得られるFine-tuningに使うデータセットの組み合わせは異なる。
- Consistency-based filterはパフォーマンス向上に貢献する。
- Negative Samplingの手法として、in-batchが一番良かった。
- BERT → RoBERTaはパフォーマンスが落ちた。
